ir/type.cpp

```diff
@@ -637,6 +637,18 @@ expr AggregateType::map_reduce(expr(*map)(const StateValue&, const StateValue&),
   return reduce(r);
 }
 
+StateValue AggregateType::aggregateVals(const vector<StateValue> &vals) const {
+  assert(defined);
```

Comment:
this assert isn't valid. `defined` is used for types that are fully defined (nothing symbolic).
So if the type is inferred, `defined` will be false, though the type will be then fine for consumption.
TL;DR: just remove this assert :)

Reply:
fixed :)